mantidproject/mantid
# Store the default dead time table in the context

**Description of work.**
This PR stores the currently active default dead time table in the corrections context to make it easier to access it from outside the corrections tab.

**To test:**
Open `Muon Analysis`
Load 84447
Go to the Corrections tab. You can see it has automatically calculated dead time `From file`.
Switch to `From workspace`. It will default to None - it should do calculation and then say 'No dead time corrections'
Choose a workspace and make sure it applies the corrections.
Switch to `From other file`. 
Try load the first data below. It should not crash, and should show an error popup.
Try load the second data below. It should select the loaded dead time workspace and calculate the dead time.

To test the `Runs` selector, load runs `500,90000` which have different dead times.
It should select the `From file` option automatically.
Select run 500 - the dead time should be zero.
Select run 90000 - the dead time should be non-zero.

Load 84447-9 and select co-add mode
The `Runs` selector should show `84447-84449` (because the corrections are performed after the summation)

Repeat for `Frequency Domain Analysis`

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
